packages/dev/core/src/Sprites/sprite.ts

```diff
@@ -213,8 +213,14 @@ export class Sprite extends ThinSprite implements IAnimatable {
         sprite.isVisible = parsedSprite.isVisible;
         sprite.useAlphaForPicking = parsedSprite.useAlphaForPicking;
 
-        sprite.fromIndex = parsedSprite.fromIndex;
-        sprite.toIndex = parsedSprite.toIndex;
+        if (parsedSprite.fromIndex < parsedSprite.toIndex) {
+            sprite.toIndex = parsedSprite.toIndex;
+            sprite.fromIndex = parsedSprite.fromIndex;
+        } else {
+            sprite.fromIndex = parsedSprite.fromIndex;
+            sprite.toIndex = parsedSprite.toIndex;
+        }

```

Comment:
I think it's hard to understand for anyone reading the code why we must do it like that. We must refer to the property  and the `playAnimation` method implementation to understand...

I would do it this way to make the code clearer (replacing lines 216-229):
```typescript
        sprite._fromIndex = parsedSprite.fromIndex;
        sprite._toIndex = parsedSprite.toIndex;
        sprite._loopAnimation = parsedSprite.loopAnimation;
        sprite._delay = parsedSprite.delay;

        sprite.playAnimation(sprite._fromIndex, sprite._toIndex, sprite._loopAnimation, sprite._delay, sprite._onAnimationEnd);

        if (parsedSprite.animationStarted) {
            sprite.playAnimation(sprite.fromIndex, sprite.toIndex, sprite.loopAnimation, sprite.delay);
        }
```
Doing it that way, we can see there's a bug, because `playAnimation` is always called, whereas there's a flag in the serialized data that can prevent automatic playing.

So, the final code would be this instead:
```typescript
        sprite._fromIndex = parsedSprite.fromIndex;
        sprite._toIndex = parsedSprite.toIndex;
        sprite._loopAnimation = parsedSprite.loopAnimation;
        sprite._delay = parsedSprite.delay;

        if (parsedSprite.animationStarted) {
            sprite.playAnimation(sprite.fromIndex, sprite.toIndex, sprite.loopAnimation, sprite.delay);
        }
```

Reply:
OK, got it, thanks!